### PR TITLE
Add MICRO_IGLU_REGISTRY_URL and MICRO_IGLU_API_KEY environment variables

### PR DESCRIPTION
**Pull request by stanch**
This offers an easy shortcut for those who don’t want to spell out a full Iglu config file.

### CODE REVIEW

File: src/main/scala/com.snowplowanalytics.snowplow.micro/ConfigHelper.scala [hunk 1]
```diff
@@ -81,32 +84,42 @@ private[micro] object ConfigHelper {
             case None => success
           }
         )
+      opt[URI]("extra-iglu-resolver")
```
Comment (istreeter) [inline src/main/scala/com.snowplowanalytics.snowplow.micro/ConfigHelper.scala:87]: To borrow [the terminology from our documentation site](https://docs.snowplow.io/docs/pipeline-components-and-applications/iglu/), I believe this is config parameter describes a "registry" not a "resolver".  A registry is the thing that hosts schemas.  A resolver is the thing that tells the client which registries to use.

**Comment by stanch** [comment]
Sorry for the noise! The thinking around how this should be configured is evolving :)

**Review by greg-el — COMMENTED** [review]
Looks good! Would be nice to update the `README` with a quick explanation/example values for the envvars (or also even in the `--help` printout? though I'm less sure if envvars belong there), in addition to the site docs

**Comment by stanch** [comment]
Last change I promise :) Added `_URL` based on feedback from Jordan.

> Would be nice to update the `README` with a quick explanation/example values for the envvars (or also even in the `--help` printout? though I'm less sure if envvars belong there), in addition to the site docs

My plan is:
* Land this and release 1.5.0
* Publish the Micro docs (https://github.com/snowplow/documentation/pull/301), which I will update to include this version, and also a new page with an overview of env vars and command line flags
* Point the README to the docs
* Add a Docker Hub README pointing to the docs

**Review by istreeter — APPROVED** [review]

**Comment by rlh1994** [comment]
Is it worth adding a check that a private registry URL ends with `/api` to avoid anyone missing that out and not knowing why it isn't working? https://docs.snowplow.io/docs/pipeline-components-and-applications/iglu/iglu-resolver/

(sorry if you already do, I couldn't see it in there but I also don't speak scala)

**Comment by stanch** [comment]
> Is it worth adding a check that a private registry URL ends with `/api` to avoid anyone missing that out and not knowing why it isn't working? https://docs.snowplow.io/docs/pipeline-components-and-applications/iglu/iglu-resolver/
> 
> (sorry if you already do, I couldn't see it in there but I also don't speak scala)

I am not sure we can/should enforce that. For example, `http://iglucentral.com` is a valid registry but does end with `/api`. Could someone theoretically run a private registry that way as well?

**Comment by rlh1994** [comment]
> > Is it worth adding a check that a private registry URL ends with `/api` to avoid anyone missing that out and not knowing why it isn't working? https://docs.snowplow.io/docs/pipeline-components-and-applications/iglu/iglu-resolver/
> > (sorry if you already do, I couldn't see it in there but I also don't speak scala)
> 
> I am not sure we can/should enforce that. For example, `http://iglucentral.com` is a valid registry but does end with `/api`. Could someone theoretically run a private registry that way as well?

I am not the person to ask that... Maybe if they provide an API key value we could at least warn/log?

**Review by greg-el — APPROVED** [review]

**Comment by stanch** [comment]
Added env vars to `--help`:
image

**Pull request merged**